services/jitpack/jitpack.service.js

```diff
@@ -31,58 +31,26 @@ module.exports = class Jitpack extends LegacyService {
           groupId: 'jitpack',
           artifactId: 'maven-simple',
         },
-        staticPreview: {
-          label: 'jitpack',
-          message: 'v1.1',
-          color: 'blue',
-        },
+        staticPreview: renderVersionBadge({ version: 'v1.1' }),
         keywords: ['java', 'maven'],
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/jitpack\/v\/([^/]*)\/([^/]*)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const groupId = `com.github.${match[1]}` // github user
-        const artifactId = match[2] // the project's name
-        const format = match[3] // "svg"
-
-        const pkg = `${groupId}/${artifactId}/latest`
-        const apiUrl = `https://jitpack.io/api/builds/${pkg}`
+  static get defaultBadgeData() {
+    return { label: 'jitpack' }
+  }
 
-        const badgeData = getBadgeData('jitpack', data)
+  async fetch({ groupId, artifactId }) {
+    return this._requestJson({
+      schema,
+      url: `https://jitpack.io/api/builds/com.github.${groupId}/${artifactId}/latest`,
+      errorMessages: { 401: 'project not found or private' },
+    })
+  }
 
-        request(apiUrl, (err, res, buffer) => {
-          if (err != null) {
-            badgeData.text[1] = 'inaccessible'
-            sendBadge(format, badgeData)
-            return
-          }
-          if (res.statusCode === 404) {
-            badgeData.text[1] = 'not found'
-            sendBadge(format, badgeData)
-            return
-          }
-          try {
-            const data = JSON.parse(buffer)
-            const status = data['status']
-            let color = versionColor(data['version'])
-            let version = versionText(data['version'])
-            if (status !== 'ok') {
```

Comment:
I don't have a clear example of when this happens, but instead of explicitly handling this, I've moved `status == 'ok'` to being enforced in the schema so we'll throw invalid response there instead. Does that seem reasonable?